### PR TITLE
AP_AHRS: send ekf status reports even when EKF inactive

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_AHRS/AP_AHRS_NavEKF.cpp [hunk 1]
```diff
@@ -1381,7 +1381,7 @@ bool AP_AHRS_NavEKF::resetHeightDatum(void)
 // send a EKF_STATUS_REPORT for current EKF
 void AP_AHRS_NavEKF::send_ekf_status_report(mavlink_channel_t chan)
 {
-    switch (active_EKF_type()) {
+    switch (ekf_type()) {
     case EKF_TYPE_NONE:
         // send zero status report
         mavlink_msg_ekf_status_report_send(chan, 0, 0, 0, 0, 0, 0);
```
Comment (OXINARF) [inline libraries/AP_AHRS/AP_AHRS_NavEKF.cpp]: Should we do nothing here instead? With this change, it means that in this case the user specifically disabled the EKF so I don't see why we would keep sending empty EKF status reports.

**Comment by peterbarker** [comment]
On Fri, 1 Sep 2017, Francisco Ferreira wrote:

> @OXINARF commented on this pull request.
> 
> I don't have anything against it, but is there a reason for the change?

Allows you to see what's happening with the EKF even if you're not using 
it.  I came across this in the APWeb interface on a Rover - the EKF fields 
weren't being filled in for no discernable reason.

Seems odd to stop telling the GCS about the estimation when it goes 
unhealthy!

> Should we do nothing here instead? With this change, it means that in this
> case the user specifically disabled the EKF so I don't see why we would keep
> sending empty EKF status reports.

Probably; I've no idea *why* we did things things way in the first place - 
it was apparently on purpose?!

**Comment by OXINARF** [comment]
> Seems odd to stop telling the GCS about the estimation when it goes
unhealthy!

It's not exactly when it goes unhealthy, but when the vehicle isn't using EKF anymore.

I can see wanting it both ways, so I don't have a formed opinion. I'll add the dev call label so we can hear other people.

> Probably; I've no idea *why* we did things things way in the first place -
it was apparently on purpose?!

I guess it would have some use before, as in, "hey, you have EKF use enabled but we stopped using it". With this change this won't happen, the only option for EKF_TYPE_NONE is when the user set the EKF not to be used, so sending a message with zeros looks dumb ("hey, you disabled the EKF use but you get the EKF message anyway").

**Comment by tridge** [comment]
looks good, thanks!

**Comment by tridge** [comment]
> hey, you disabled the EKF use but you get the EKF message anyway

I'm happy with the change as that is an unusual case - we default to using EKF on all vehicle types. We could skip sending the messages I guess, so it would be fine as an additional change.
I merged the change as this is a real bug for rover and plane when they fall back to DCM

**Pull request merged**